contribs/gnodev/main.go

```diff
@@ -379,6 +436,29 @@ func checkForError(w io.Writer, err error) {
 	fmt.Fprintln(w, "[DONE]")
 }
 
+func resolveUnixOrTCPAddr(in string) (out string) {
```

Comment:
Why do we need this validation if it's done by bft anyway?

Reply:
It's not really for validation but for convenience.
The node resolves the address by requiring the full address format `tcp://<addr>:port`. I created this method to allow specifying `-node-rpc-listener=:8282` instead of `-node-rpc-listener=tcp://0.0.0.0:8282`.
If you prefer, I can update this method to only resolve TCP addresses and fallback to returning the input address, allowing the node to handle other cases like UDS.

Reply:
Also, pre-resolving a random port `:0` prevents the node from receiving a new random port upon each reload (I guess?). This modification will avoid gnoweb from failing until I reintroduce a mechanism to update the remote node address during reload.